File: CpRust/src/replace.rs
use std::io::stdin;

fn _rin() -> Vec<char> {
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    input.trim().chars().collect()
}

fn replace(orgin: &mut Vec<char>, target: Vec<char>, rep: Vec<char>) {
    let n = orgin.len();
    let t_len = target.len();
    let r_len = rep.len();

    let mut i = 0;
    while i <= n - t_len {
        if orgin[i..i + t_len] == target[..] {
            orgin.splice(i..i + t_len, rep.iter().cloned());
            i += r_len;
        } else {
            i += 1;
        }
    }
}

fn main() {
    let mut orgin = _rin();
    let target = _rin();
    let rep = _rin();
    replace(&mut orgin, target, rep);
    println!("{}", orgin.iter().collect::<String>());
}

File: CpRust/src/replace.cpp
#include <iostream>
#include <string>
using namespace std;

void replace(string &origin, string &target, string &rep) {
  int n = origin.length();
  for (int i = 0; i < n; i++) {
    if (origin[i] == target[0]) {
      bool match = true;
      for (int j = 0; j < target.size(); j++) {
        if (target[j] != origin[i + j]) {
          match = false;
          break;
        }
      }
      if (match) {
        for (int j = 0; j < target.size(); j++) {
          origin[i + j] = rep[j];
        }
      }
    }
  }
}

int main(int argc, char *argv[]) {
  string origin, target, rep;
  cin >> origin >> target >> rep;

  replace(origin, target, rep);
  std::cout << origin << '\n';
  return 0;
}
